modules/magnolia/src/main/scala/pureconfig/module/magnolia/ExportedMagnolia.scala

```diff
@@ -0,0 +1,24 @@
+package pureconfig.module.magnolia
+
+import magnolia.Magnolia
+import pureconfig.Exported
+
+import scala.language.higherKinds
+import scala.reflect.macros.{ blackbox, whitebox }
+
+// Wrap the output of Magnolia in an Exported to force it to a lower priority.
+// This seems to work, despite magnolia hardcode checks for `macroApplication` symbol
+// and relying on getting an diverging implicit expansion error for auto-mode.
+// Thankfully at least it doesn't check the output type of its `macroApplication`
+object ExportedMagnolia {
+  def exportedMagnolia[TC[_], A](c: blackbox.Context)(implicit t: c.WeakTypeTag[A]): c.Expr[Exported[TC[A]]] = {
+    val magnoliaTree = forcedBlackboxMagnolia[TC, A](c)
+    c.universe.reify(Exported(magnoliaTree.splice))
+  }
+
+  // if i TELL you it's whitebox, you BETTER believe me
+  def forcedBlackboxMagnolia[TC[_], A](c: blackbox.Context)(implicit t: c.WeakTypeTag[A]): c.Expr[TC[A]] = {
+    val c0 = c.asInstanceOf[whitebox.Context]
```

Comment:
This is safe because the actual Context instance always implements both blackbox & whitebox interfaces:

```scala
abstract class Context extends scala.reflect.macros.blackbox.Context
                          with scala.reflect.macros.whitebox.Context
```